main.go

```diff
@@ -3,51 +3,65 @@ package main
 import (
 	"log"
 	"os"
-	"os/signal"
-	"strings"
 
 	"github.com/Shopify/sarama"
-
-	"github.com/digicatapult/wasp-ingest-rtmp/services"
-	"github.com/digicatapult/wasp-ingest-rtmp/util"
+	"github.com/pkg/errors"
+	"go.uber.org/zap"
 )
 
-func setupProducer(kafkaBrokers []string) (sarama.AsyncProducer, error) {
-	return sarama.NewAsyncProducer(kafkaBrokers, nil)
+func setupProducer(kafkaBrokers []string) (sarama.SyncProducer, error) {
+	var (
+		producer sarama.SyncProducer
+		err      error
+	)
+
+	if producer, err = sarama.NewSyncProducer(kafkaBrokers, nil); err != nil {
+		return producer, errors.Wrap(err, "problem initiating producer")
+	}
+
+	return producer, nil
 }
 
 func main() {
-	sarama.Logger = log.New(os.Stdout, "[Sarama] ", log.LstdFlags)
-
-	kafkaBrokers := util.GetEnv(util.KafkaBrokersEnv, "localhost:9092")
-	producer, errProducer := setupProducer(strings.Split(kafkaBrokers, ","))
-	if errProducer != nil {
-		panic(errProducer)
+	cfg := zap.NewDevelopmentConfig()
+	if os.Getenv("ENV") == "production" {
+		cfg = zap.NewProductionConfig()
+		lvl, err := zap.ParseAtomicLevel(os.Getenv("LOG_LEVEL"))
+		if err != nil {
+			panic("invalid log level")
+		}
+		log.Printf("setting level: %s\n", lvl.String())
+		cfg.Level = lvl
+	}
+	logger, err := cfg.Build()
+	if err != nil {
+		panic("error initialising the logger")
 	}
+	defer logger.Sync()
+	zap.ReplaceGlobals(logger)
 
-	defer func() {
-		if err := producer.Close(); err != nil {
-			log.Fatalln(err)
-		}
-	}()
+	zap.S().Debug("DEBUG")
+	zap.S().Info("INFO")
+	zap.S().Warn("WARN")
+	zap.S().Error("ERROR")
 
-	// Trap SIGINT to trigger a graceful shutdown.
-	signals := make(chan os.Signal, 1)
-	signal.Notify(signals, os.Interrupt)
+	// sarama.Logger = log.New(os.Stdout, "[Sarama] ", log.LstdFlags)
 
-	kafka := services.NewKafkaService(producer)
+	// kafkaBrokers := util.GetEnv(util.KafkaBrokersEnv, "localhost:9092")
 
-	messageKey := "01000000-0000-4000-8883-c7df300514ed"
-	messageValue := services.KafkaMessage{
-		Ingest:    "rtmp",
-		IngestID:  "4883C7DF300514ED",
-		Timestamp: "2021-08-31T14:51:36.507Z",
-		Payload:   "",
-		Metadata:  "{}",
-	}
+	// producer, errProducer := setupProducer(strings.Split(kafkaBrokers, ","))
+	// if errProducer != nil {
+	// 	panic(errProducer)
+	// }
+
+	// defer func() {
+	// 	if err := producer.Close(); err != nil {
+	// 		log.Fatalln(err)
+	// 	}
+	// }()
 
-	kafka.SendMessage(messageKey, messageValue, signals)
+	// kafka := services.NewKafkaService(producer)
 
-	// videoIngest := services.NewVideoIngestService()
+	// videoIngest := services.NewVideoIngestService(kafka)
 	// videoIngest.IngestVideo()
```

Comment:
Is this meant to be commented?

Reply:
should be updated, but will also need the rtmp url branch first